### PR TITLE
fix(android/app) Download associated dictionary when installing cloud keyboard package

### PR DESCRIPTION
**Pull request by darcywong00**
When installing cloud keyboard packages that have a specified language ID, this PR fixes the feature to download the associated dictionary package.

Some things to note about the current staging keyboard search:
* Keyboard searches won't specify language ID unless it's a language query (`q:l:id:...`)
* Search results currently use `tag=` for language ID, but eventually production will use `bcp47=`.
* Language searches involving script (e.g. `glk-arab`) currently return a tag `glk`. If the associated dictionary package (`sil.glk-arab.ptwl1`) doesn't include that tag (it only lists `glk-arab`), the dictionary package won't get installed.

### User Testing
@MakaraSok 
1. Load the test build
2. Install fv_sencoten from the cloud
    * From "Settings" --> "Install Keyboard or Dictionary" --> "Install from keyman.com"
    * In the search bar, do a language search for SENĆOŦEN by typing `l:id:str`
    Note, make sure the system keyboard doesn't change your query with autocorrection (Gboard would replace `id` with `I'd`)
Screenshot_1593741487
    * Install fv_sencoten keyboard package
3. Keyman will find the associated `nrc.str.sencoten` dictionary package and install it in the background (this may take some time depending on network traffic)
- [ ] Verify Keyman installs SENĆOŦEN keyboard and dictionary packages (keyboard displays **Straits Salish (Latin)**

### CODE REVIEW

**Comment by mcdurdin** [comment]
https://build.palaso.org/viewLog.html?buildId=214002&buildTypeId=KeymanAndroid_TestSamplesAndTestProjects&tab=buildLog seems to have failed although it hasn't reported back here.

**Comment by darcywong00** [comment]
argh, it's an unset variable that happens on release builds. Will put a 🔥 PR soon

File: android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java [hunk 1]
```diff
@@ -609,9 +615,12 @@ private void downloadKMP(String scheme) {
           FileUtils.getFilename(url);
         }
 
-        // Parse the url for the BCP 47 language ID
-        Uri uri = Uri.parse(url);
-        String languageID = uri.getQueryParameter(KMKeyboardDownloaderActivity.KMKey_BCP47);
+        // Parse data for the BCP 47 language ID
+        String languageID = data.getQueryParameter(KMKeyboardDownloaderActivity.KMKey_BCP47);
```
Comment (mcdurdin) [inline android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java:619]: If `bcp47` (and `tag`) are not specified, will this work? Just checking that the parameter is optional.

Comment (darcywong00) [reply android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java:619]: yep, `getQueryParameter()` returns `null if the parameter isn't found.

**Comment by MakaraSok** [comment]
@Darcy, Tested with keyman-14.0.108.apk alpha (released 2020-07-09, 6.12 MB) on Nexus 5 running Android 6.0.1:

- The keyboard does not get installed after tapping on the green button "Install keyboard". I was instead taken to this page:
install keyboard on android

- Tapping on "2. Install \<keyboard name\>" took me back to the keyboard home page without installing the keyboard nor the associate dictionary package:
keyboard homepage

**Comment by darcywong00** [comment]
ah, the alpha builds currently need to finish keyboard install integration (universal links)  this sprint.
It's separate from this PR, so I'll notify when the build is ready to resume testing.

**Comment by MakaraSok** [comment]
Tested again and the model got installed about the same time the keyboard was installed. The test was done on Android 10.

image

**Pull request merged**